lib/APPRemoteControl/App.cpp

```diff
@@ -249,50 +324,44 @@ void App::reportVehicleData() const
  */
 static void App_cmdChannelCallback(const uint8_t* payload, const uint8_t payloadSize, void* userData)
 {
-    (void)userData;
-    if ((nullptr != payload) && (sizeof(RemoteCtrlState::CmdId) == payloadSize))
+    if ((nullptr != payload) && (COMMAND_CHANNEL_DLC == payloadSize) && (nullptr != userData))
     {
-        RemoteCtrlState::CmdId cmdId = *reinterpret_cast<const RemoteCtrlState::CmdId*>(payload);
-
-        RemoteCtrlState::getInstance().execute(cmdId);
+        App*          application = reinterpret_cast<App*>(userData);
+        const Command cmd         = *reinterpret_cast<const Command*>(payload);
+        application->handleRemoteCommand(cmd);
     }
```

Comment:
I suggest a log message in the else case as this if may fail during development cycles and a log would help debugging.